src/masoniteorm/query/QueryBuilder.py

```diff
@@ -1294,6 +1295,22 @@ def first(self, query=False):
 
         return self.prepare_result(result)
 
+    def sole(self, query=False):
+        """Gets the only record matching a given criteria.
+
+        Returns:
+            dictionary -- Returns a dictionary of results.
+        """
+        result = self.take(2).get()
+
+        if result.is_empty():
+            raise NoRecordsFound()
```

Comment:
Going to change this to ModelNotFound to more match another PR going in for find_or_fail